api/tests/core/users/test_models.py

```diff
@@ -22,11 +22,12 @@
 from pcapi.models import db
 from pcapi.models.validation_status_mixin import ValidationStatus
 from pcapi.repository import repository
+from pcapi.repository import transaction
 
-
-pytestmark = pytest.mark.usefixtures("db_session")
+from tests.conftest import clean_database
 
 
+@pytest.mark.usefixtures("db_session")
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏Pourquoi tu déplace la fixture au niveau des fonctions ? 
`pytestmark = pytest.mark.usefixtures("db_session")` devrait avoir le même comportement

Reply:
C'est parce que la fixture `db_session` interfère avec le cycle de transaction ce qui empêche de tester le trigger `ensure_password_or_sso_exists`

Le trigger n'est déclenché qu'à la fin de la transaction, parce qu'il est `defferable initially deffered` , et la transaction de `db_session` ne se termine qu'après le test. Tu peux tester ce comportement en remplaçant `clean_database` par `db_session` dans [le test d'insertion qui doit lever une exception](https://github.com/pass-culture/pass-culture-main/pull/9274/files#diff-d86ae262237937e1f32fdaa7d65e7ff2998aa722a416eeedbcc3ecf8df03dbecR534)

Reply:
C'est assez spécifique pour mériter un commentaire dans le code je pense :smile: